app/gui/src/project-view/components/widgets/FileBrowserWidget/nameBar.ts

```diff
@@ -20,8 +20,13 @@ export function useNameBar() {
     fileExtensionFilter,
     setFilename: (filename: string) => {
       const [name, extension] = splitFilename(filename)
-      filenameInput.value = name
-      extensionInput.value = extension
+      // Filtering datalinks does not make much sense, and we should probably change behavior for other extensions as well in the future.
+      if (extension === 'datalink') {
+        filenameInput.value = filename
```

Comment:
Should `extensionInput` be cleared here?

Reply:
I think not, because it provides a slightly better user experience without unnecessary changes to filtering. However, the entire filtering system, by extension, can be significantly improved.